drivers/bluetooth/hci/ipm_stm32wb.c

```diff
@@ -96,6 +96,37 @@ static void syscmd_status_not(SHCI_TL_CmdStatus_t status)
 	BT_DBG("status:%d", status);
 }
 
+/*
+ * https://github.com/zephyrproject-rtos/zephyr/issues/19509
+ * Tested on nucleo_wb55rg (stm32wb55rg) BLE stack (v1.2.0)
+ * Unresolved Resolvable Private Addresses (RPA)
+ * is reported in the peer_rpa field, and not in the peer address,
+ * as it should, when this happens the peer address is set to all FFs
+ * 0A 00 01 08 01 01 FF FF FF FF FF FF 00 00 00 00 00 00 0C AA C5 B3 3D 6B ...
+ * If such message is passed to HCI core than pairing will essentially fail.
+ * Solution: Rewrite the event with the RPA in the PEER address field
+ */
+static void tryfix_event(TL_Evt_t *tev)
+{
+	struct bt_hci_evt_le_meta_event *mev =
+			(struct bt_hci_evt_le_meta_event *)&tev->payload;
+	u8_t *ptr = ((u8_t *)mev + (sizeof(*mev)));
```

Comment:
This `ptr` variable seems a bit unnecessary - how come you don't just assign directly to `evt`? Even if you did have an intermediate variable wouldn't `void *` be better since it doesn't require explicit typecasting? (I'd still go for just dropping this variable however)

Reply:
@pavlohamov btw, thanks for your patience with the reviews. Once this last small thing is sorted I think we can go and merge this.

Reply:
It's just a matter of fitting in one line. for one cast from ptr is much more clear than 
```
struct bt_hci_evt_le_enh_conn_complete *evt =
		(struct bt_hci_evt_le_enh_conn_complete *)((u8_t *)mev +
		(sizeof(*mev)));
```
No problem. Thank you for the pedantic review. During review I've learned couple of thing about zephyr HCI core. Now I understand the internals of HCI packet. at least one =)

Reply:
You could do a `(void *)` cast to make the line shorter. It's equally valid (as in, the compiler won't complain) as the exact type on the left side of `=`

Reply:
Make sense. Now two extra lines are saved. Thanks,